tests/README.rst

```diff
@@ -80,6 +80,153 @@ To do that, run `podman pull quay.io/rhinstaller/anaconda-ci:master` or build
+
+
+This could be potentially insecure so we’re trying what we can to make it secure.
+We applied a lot of hardening which I would like to describe here.
+
+Getting into our host/internal network
+______________________________________
+
+One of the main precautions is that each container running test will run only
+a limited time and is destroyed after timeout/end of test. That should narrow
+what attackers could do or how they can create a backdoor. See the image for
+more info:
+
+.. image:: ../docs/images/tests/GH-self-hosted-runners.png
+
+
+Another hardening of this is potential issue is that PRs only
+approved by/created by users with permission to write are able to run the tests.
+To achieve this we have two ways how to start the test.
+
+**PR created by user** -- these are started from the RHEL branch workflow file by
+`pull_request_target` as usual. This workflow has two dependent jobs. First
+will check user privileges, second will run the tests in case the first one
+succeeded.
+
+**PR created by external contributors** -- these have to be started by workflow
+file from the `master` branch checking all the comments. If comment have
+`/test` phrase it will check the owner of the comment. When everything succeed
+it will set progress on the pull request originating the comment and start the
+tests. This progress is updated based on the result of the tests.
+As explained above, the whole implementation of the workflow is in the `master` branch
+which could be pretty confusing.
+
+Changing workflow file by attacker
+__________________________________
+
+Because test description is part of the repository attackers may change
+workflow files by creating PR to do their malicious attack. This could normally
+happen. Because of that we are using `pull_request_target` instead of
+`pull_request` trigger. The main difference is that `pull_request_target`
+will run your PR tests on the target branch not on your PR branch. So workflow
+configuration has to be merged first to apply workflow changes.
+This has to be set on all workflow files in all branches otherwise we are
+vulnerable because attackers could change existing workflow files to use our
+runners even for branches where they are not normally used. Unfortunately,
+self-hosted runners can’t be bound to the branch, they are bound to the repo.
+
+How can I change the configuration
+__________________________________
+
+Thanks to our hardening it’s not possible to just create PR and see the result
+of your change on the PR checks tab. You have to create PR on your fork branch
+which has the updated workflow. I would recommend you to create a test
+organization for this and avoid creating a new account.
+
+Similar situation works even for workflow to automatically update our containers.
+This workflow has `schedule` and `manual_dispatch` triggers which are always
+started on the default branch. For that you can change the default branch on your
+fork repository.
```

Comment:
"has.. triggers. `schedule` triggers are always run on the default branch. For testing updates, always add `manual_dispatch` so that you can run them from your branch (on either origin or your fork).